Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -269,6 +271,88 @@ lemma iff_lTensor_preserves_injective_linearMap [Small.{v} R] :
       (L : N →ₗ[R] N'), Function.Injective L → Function.Injective (L.lTensor M) := by
   simp_rw [iff_rTensor_preserves_injective_linearMap, LinearMap.lTensor_inj_iff_rTensor_inj]
 
+variable (R M) in
+lemma lTensor_exact [Small.{v} R] [flat : Flat R M] ⦃N N' N'' : Type v⦄
+    [AddCommGroup N] [AddCommGroup N'] [AddCommGroup N''] [Module R N] [Module R N'] [Module R N'']
+    (f : N →ₗ[R] N') (g : N' →ₗ[R] N'') (exact : Function.Exact f g) :
+    Function.Exact (f.lTensor M) (g.lTensor M) := by
+  let π : N' →ₗ[R] N' ⧸ LinearMap.range f := Submodule.mkQ _
+  have exact0 : Function.Exact f π := by intro x; simp [π]
+  have surj0 : Function.Surjective π := Quotient.surjective_Quotient_mk''
+
+  let ι : N' ⧸ LinearMap.range f →ₗ[R] N'' :=
+    Submodule.subtype _ ∘ₗ (LinearMap.quotKerEquivRange g).toLinearMap ∘ₗ
+      Submodule.quotEquivOfEq (LinearMap.range f) (LinearMap.ker g)
+        (LinearMap.exact_iff.mp exact).symm
+  have inj0 : Function.Injective ι := by simpa [ι] using Subtype.val_injective
+  have eq0 : g = ι.comp π := by aesop
+
+  suffices exact1 : Function.Exact (f.lTensor M) (π.lTensor M) by
+    rw [eq0, lTensor_comp]
+    apply Function.Exact.comp_injective (exact := exact1)
+      (inj := iff_lTensor_preserves_injective_linearMap R M |>.mp flat _ inj0)
+      (h0 := map_zero _)
+
+  exact _root_.lTensor_exact _ exact0 surj0
+
+variable (R M) in
+lemma rTensor_exact [Small.{v} R] [flat : Flat R M] ⦃N N' N'' : Type v⦄
+    [AddCommGroup N] [AddCommGroup N'] [AddCommGroup N''] [Module R N] [Module R N'] [Module R N'']
+    (f : N →ₗ[R] N') (g : N' →ₗ[R] N'') (exact : Function.Exact f g) :
+    Function.Exact (f.rTensor M) (g.rTensor M) := by
+  let π : N' →ₗ[R] N' ⧸ LinearMap.range f := Submodule.mkQ _
+  have exact0 : Function.Exact f π := by intro x; simp [π]
+  have surj0 : Function.Surjective π := Quotient.surjective_Quotient_mk''
+
+  let ι : N' ⧸ LinearMap.range f →ₗ[R] N'' :=
+    Submodule.subtype _ ∘ₗ (LinearMap.quotKerEquivRange g).toLinearMap ∘ₗ
+      Submodule.quotEquivOfEq (LinearMap.range f) (LinearMap.ker g)
+        (LinearMap.exact_iff.mp exact).symm
+  have inj0 : Function.Injective ι := by simpa [ι] using Subtype.val_injective
+  have eq0 : g = ι.comp π := by aesop
+
+  suffices exact1 : Function.Exact (f.rTensor M) (π.rTensor M) by
+    rw [eq0, rTensor_comp]
+    apply Function.Exact.comp_injective (exact := exact1)
+      (inj := iff_rTensor_preserves_injective_linearMap R M |>.mp flat _ inj0)
+      (h0 := map_zero _)
+
```

Comment:
We could reuse the proof in `lTensor_exact`, but I think it is faster to just copy and paste